### DISCUSSION TITLE
Official EKS Add-On Status

### DISCUSSION THREAD

**Discussion by stalbot15** [aws/secrets-store-csi-driver-provider-aws#402] [General]
Similar to the [aws-ebs-csi-driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver), this driver is currently packaged as a helm chart. Unlike the EBS driver, however, this one is not an [official EKS add-on](https://docs.aws.amazon.com/eks/latest/userguide/workloads-add-ons-available-eks.html). Is there a reason this couldn't be promoted as an official add-on?

**Comment by simonmarty** [comment]
We are discussing this internally.

There are multiple ways to consume secrets (directly in your application, with one of our [client-side caching libraries](https://aws.amazon.com/blogs/security/use-aws-secrets-manager-client-side-caching-libraries-to-improve-the-availability-and-latency-of-using-your-secrets/), through the [CSI driver](https://github.com/aws/secrets-store-csi-driver-provider-aws), through a 3rd-party secrets provider, with our newly released [Secrets Manager Agent](https://github.com/aws/aws-secretsmanager-agent) etc). Each solution has its advantages and inconvenients. A big difference between these solutions is how they affect the the size of the threat boundary beyond which accessing the secret value becomes trivial (with fetching your secret directly within the application  being safest in that regard). It's a delicate balance between convenience and security. 

The case for EBS is a lot more straightforward imo since a CSI driver is the obvious solution for attaching a block store to a k8s application.

If you do find pain points in installing the secrets store CSI driver and our AWS provider, do let us know by opening an issue.
